### PR TITLE
[BACKUP] `az backup policy`: Add support for Smart Tiering policy

### PR DESCRIPTION
**Pull request by akshayneema**
**Related command**

**Description**

`az backup policy create/set`: Add support for creating and updating backup policies with smart tiering enabled 
`az backup policy list`: Add `--move-to-archive-tier` parameter to selectively list policies with smart tiering enabled/disabled

**Testing Guide**

**History Notes**

`az backup policy create/set`: Add support for creating and updating backup policies with smart tiering enabled 
`az backup policy list`: Add `--move-to-archive-tier` parameter to selectively list policies with smart tiering enabled/disabled

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
BACKUP

File: src/azure-cli/requirements.py3.windows.txt [hunk 1]
```diff
@@ -33,7 +33,7 @@ azure-mgmt-consumption==2.0.0
 azure-mgmt-containerinstance==9.1.0
 azure-mgmt-containerregistry==10.0.0
 azure-mgmt-containerservice==20.2.0
-azure-mgmt-core==1.3.1
+azure-mgmt-core==1.3.2
```
Comment (zhoxing-ms) [inline src/azure-cli/requirements.py3.windows.txt:36]: May I ask why do you need to upgrade the package version for `azure-mgmt-core` and `msrest` in this PR?

Comment (akshayneema) [reply src/azure-cli/requirements.py3.windows.txt:36]: The pipelines were failing without these changes. Our new recoverservicesbackup SDK has dependencies on these versions -

[](https://github.com/Azure/azure-sdk-for-python/blob/d8c47ab85b0f97e8d51769be1ff63e8faa940723/sdk/recoveryservices/azure-mgmt-recoveryservicesbackup/setup.py#L72-L74)

Comment (msyyc) [reply src/azure-cli/requirements.py3.windows.txt:36]: `azure-mgmt-core==1.3.2` only upgrade dependency for `azure-core` which has been upgraded in azure-cli. It does not have new feature so there is no influence for azure-cli.

Comment (zhoxing-ms) [reply src/azure-cli/requirements.py3.windows.txt:36]: Thank you for helping clarify this upgrade

In addition, I can see from the history of these packages that the impact of their upgrade should not be big:

For https://pypi.org/project/azure-mgmt-core/1.3.2/
```
Updated mindep about azure-core from 1.23.0 to 1.24.0
```
`1.24.0` is currently the dependent version of CLI

For https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/requirements.py3.Darwin.txt#L11

For https://pypi.org/project/msrest/
```
Declare correctly msrest as Python 3.6 and more only for clarity #251
```
Related issue: https://github.com/Azure/msrest-for-python/issues/251

**Comment by zhoxing-ms** [comment]
@akshayneema Could you please resolve these conflicts~

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**